velox/functions/lib/Re2Functions.cpp

```diff
@@ -374,11 +424,65 @@ class Re2SearchAndExtract final : public VectorFunction {
   const bool emptyNoMatch_;
 };
 
+// Match string 'input' with a fixed pattern (with no wildcard characters).
+bool matchExactPattern(
+    StringView input,
+    StringView pattern,
+    vector_size_t length) {
+  return input.size() == pattern.size() &&
+      std::memcmp(input.data(), pattern.data(), length) == 0;
+}
+
+// Match the first 'length' characters of string 'input' and prefix pattern.
+bool matchPrefixPattern(
+    StringView input,
+    StringView pattern,
+    vector_size_t length) {
+  return input.size() >= length &&
+      std::memcmp(input.data(), pattern.data(), length) == 0;
+}
+
+// Match the last 'length' characters of string 'input' and suffix pattern.
+bool matchSuffixPattern(
+    StringView input,
+    StringView pattern,
+    vector_size_t length) {
+  return input.size() >= length &&
+      std::memcmp(
+          input.data() + input.size() - length,
+          pattern.data() + pattern.size() - length,
+          length) == 0;
+}
+
+template <PatternKind P>
 class LikeConstantPattern final : public VectorFunction {
  public:
-  LikeConstantPattern(StringView pattern, std::optional<char> escapeChar)
+  LikeConstantPattern(
+      StringView pattern,
+      vector_size_t reducedPatternLength,
+      std::optional<char> escapeChar)
       : re_(toStringPiece(likePatternToRe2(pattern, escapeChar, validPattern_)),
```

Comment:
We should not compile RE2 for simple patterns. Let's split this class into 2. Use one for kGeneric pattern and the other for the rest of the patterns.

Reply:
Sure, thank you!